graphqlapi/src/main/java/io/stargate/graphql/schema/types/TupleBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.types;
+
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TupleBuilder extends TypeBuilder {
+  private final List<GraphQLType> subTypes;
+
+  public TupleBuilder(List<GraphQLType> subTypes) {
+    this.subTypes = subTypes;
+  }
+
+  private String getName(boolean isInput) {
+    return String.format(
+        "Tuple%s%s",
+        subTypes.stream().map(this::getTypeName).collect(Collectors.joining()),
+        isInput ? "Input" : "");
+  }
+
+  public GraphQLOutputType buildOutputType() {
+    List<GraphQLFieldDefinition> fields = new ArrayList<>(subTypes.size());
+
+    for (int i = 0; i < subTypes.size(); i++) {
+      GraphQLOutputType t = (GraphQLOutputType) subTypes.get(i);
+      fields.add(
+          GraphQLFieldDefinition.newFieldDefinition()
+              .name(String.format("item%d", i))
+              .type(i == 0 ? new GraphQLNonNull(t) : t)
+              .build());
+    }
+
+    return GraphQLObjectType.newObject()
+        .description(String.format("Represents a Tuple type with %d items", subTypes.size()))
```

Comment:
If we're going to mention the origin type maybe we should describe it fully, something like `Represents an <int,int> tuple`.

Reply:
hm... I wanted to avoid showing the CQL types in the description. The description is a meta property of the schema and it can be viewed along side the GraphQL fields, so there shouldn't be a confusion about what those "n times" are.

Reply:
So then do we need the description at all :) It's pretty easy to count the number of fields in the schema.

Reply:
I think it can be helpful, specially for complex types like `TupleIntTupleTextText`. This description can be viewed quickly as a tooltip on IDEs.

In any case, I don't feel very strongly about it, I can remove this part of the description.